### PR TITLE
Improved lastname anonymization

### LINKED ISSUE #30345: Problems with lastname anonymization in the productcomments module
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If you have a customer account with some Latin characters like ąśćł and an anonymization setting enabled, the comments won't load after adding one from such an account.

Second problem: anonymization doesn't work for guest accounts.

### Expected behavior

The comments should load properly. Anonymization should work for the guest accounts.

### Steps to reproduce

0. Add a comment as a customer with "John Doe" as a name, validate it, and see that it is correctly displayed.
1. Create a customer account: "Jan Łomża"
2. Enable anonymization in the product comments module.
3. Add a comment.
4. Accept it
5. See that comments are not loading anymore.

### PrestaShop version(s) where the bug happened

1.7.8.7 - develop - 8.0.0
### PHP version(s) where the bug happened

n/a

### If your bug is related to a module, specify its name and its version

productcomments v5.0.2

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Anonymization now works for guests and customers
| Type?         | bug fix
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#30345.
| How to test?  | Please indicate how to best verify that this PR is correct.
| Sponsor company | impSolutions

### CODE REVIEW

File: controllers/front/ListComments.php
Comment (leemyongpakvn) [inline controllers/front/ListComments.php:76]: I think `$productComment['firstname'] = $parts[0];` will be better.
Because this variable is used only one time and its name looks confused.

**Review by matks — COMMENTED** [review]
Code seems ok but I think we need to help developers to understand this new logic. Maybe add a comment or create a private method with a very explicit name?

Also PHPStan is not happy
```
 -- ------------------------------------------------------------------ 
     Error                                                             
 -- ------------------------------------------------------------------ 
     Ignored error pattern #Access to an undefined property            
     Cookie::\$id_customer.# was not matched in reported errors.       
     Ignored error pattern #Access to an undefined property            
     Cookie::\$id_guest.# was not matched in reported errors.          
     Ignored error pattern #Access to an undefined property            
     HelperList::\$list_id.# was not matched in reported errors.       
     Ignored error pattern #Access to an undefined property            
     HelperList::\$shopLinkType.# was not matched in reported errors.  
 -- ------------------------------------------------------------------ 
 ```

**Review by leemyongpakvn — COMMENTED** [review]
as old code has 1 `isset` check at line 67

File: controllers/front/ListComments.php
Comment (kpodemski) [reply controllers/front/ListComments.php]: thanks @leemyongpakvn, you're right :))

**Review by Hlavtox — APPROVED** [review]

File: controllers/front/ListComments.php
Comment (micka-fdz) [inline controllers/front/ListComments.php:76]: I think you used wrong variable here.
```suggestion
            $productComment['customer_name'] = htmlentities($productComment['customer_name']);
```

**Review by micka-fdz — CHANGES_REQUESTED** [review]

File: controllers/front/ListComments.php
Comment (kpodemski) [reply controllers/front/ListComments.php]: @micka-fdz nice catch!

**Comment by HanaRebaiQA** [comment]
Hello @kpodemski 

Thanks for your PR. Can we have a second review before starting testing following @micka-fdz 'feedback review ?

**Review by micka-fdz — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Review by MhiriFaten — APPROVED** [review]
Hello @kpodemski ,

I have checked your PR, the issue is well fixed!

[Untitled_ Dec 9, 2022 3_33 PM.webm](https://user-images.githubusercontent.com/111756615/206725376-3df4fbaf-3aee-4d63-b5fa-4f1973ee26e5.webm)

So it's QA approved :heavy_check_mark: 

Many thanks !

**Comment by Hlavtox** [comment]
Thank you @kpodemski!!!!

**Pull request merged**
